federationsender/storage/shared/storage_edus.go

```diff
@@ -111,6 +120,7 @@ func (d *Database) CleanEDUs(
 			}
 			if count == 0 {
 				deleteNIDs = append(deleteNIDs, nid)
+				d.Cache.EvictFederationSenderQueuedEDU(nid)
```

Comment:
So this gets called when _all_ servers have been sent this EDU?

Reply:
Yep, exactly. It's just a way of cleaning up the cache a bit rather than leaving it there until it's evicted by something else.